app/Services/BuslineUserService.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use App\User;
+
+class BuslineUserService
+{
+    /**
+     * Attaches Busline id to User model.
+     */
+    public function subscribeBusline($busline): void
+    {
+        $user = auth()->user();
+        $user->buslines()->syncWithoutDetaching($busline);
+    }
+
+    /**
+     * Detaches Busline id from User model.
+     */
+    public function unsubscribeBusline($busline): void
+    {
+        $user = auth()->user();
+        $user->buslines()->detach($busline);
+    }
+
+    /**
+     * Shows Buslines subscribed by User.
+     */
+    public function userSubscribedBuslines(): array
+    {
+        $user = auth()->user();
```

Comment:
Try to avoid using helpers like `auth()->user()` in places like this. Why service would need to know how to find logged user? It should be passed to it via method parameter. I'll be easier to test too in that case.

Reply:
Best way to fetch user in controller will be taking it from `Request` class, I think. Using facades like `Auth::user()` or helpers can be problematic in future.